app/uk/gov/hmrc/nrs/retrieval/controllers/NonrepRetrievalController.scala

```diff
@@ -19,61 +19,70 @@ package uk.gov.hmrc.nrs.retrieval.controllers
 import javax.inject.Singleton
 import com.google.inject.Inject
 import play.api.mvc._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.http.{BadGatewayException, HeaderCarrier, HttpResponse, InternalServerException, NotFoundException}
 import uk.gov.hmrc.nrs.retrieval.connectors.NonrepRetrievalConnector
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
 import scala.concurrent.ExecutionContext
 
 @Singleton()
-class NonrepRetrievalController @Inject()(val nonrepRetrievalConnector: NonrepRetrievalConnector,
-                                          override val controllerComponents: ControllerComponents)
-                                         (implicit ec: ExecutionContext)
-  extends BackendController(controllerComponents) {
+class NonrepRetrievalController @Inject()(
+  val nonrepRetrievalConnector: NonrepRetrievalConnector,
+  override val controllerComponents: ControllerComponents)(implicit ec: ExecutionContext)
+    extends BackendController(controllerComponents) {
 
-   def search() = Action.async { implicit request =>
+  val search: Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.search(mapToSeq(request.queryString)).map(response => Ok(response.body))
   }
 
-  def submitRetrievalRequest(vaultId: String, archiveId: String) = Action.async { implicit request =>
+  def submitRetrievalRequest(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.submitRetrievalRequest(vaultId, archiveId).map(response => rewriteResponse(response))
   }
 
-  def statusSubmissionBundle(vaultId: String, archiveId: String) = Action.async { implicit request =>
+  def statusSubmissionBundle(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.statusSubmissionBundle(vaultId, archiveId).map(response => rewriteResponse(response))
   }
 
-  def getSubmissionBundle(vaultId: String, archiveId: String) = Action.async { implicit request =>
-    nonrepRetrievalConnector.getSubmissionBundle(vaultId, archiveId)
-      .map{response => Ok(response.bodyAsBytes).withHeaders(mapToSeq(response.headers):_*)}
+  def getSubmissionBundle(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
+    nonrepRetrievalConnector.getSubmissionBundle(vaultId, archiveId).map { response =>
+      val errorMessage =
+        s"get submission bundle for vaultId: [$vaultId] archiveId: [$archiveId] received unexpected response status: ${response.status.toString}"
+
+      response.status match {
+        case NOT_FOUND                                 => throw new NotFoundException(errorMessage)
+        case status if status >= INTERNAL_SERVER_ERROR => throw new BadGatewayException(errorMessage)
+        case status if status >= MULTIPLE_CHOICES      => throw new InternalServerException(errorMessage)
```

Comment:
I find this part confusing.
Because of throwing exceptions and the way how status code is treated, 
e.g. status >= INTERNAL_SERVER_ERROR is thrown as bad gateway
and anything >= 300 as internal server exception